x-pack/plugins/observability_ai_assistant/server/service/types.ts

```diff
@@ -15,13 +14,15 @@ import type {
   KnowledgeBaseEntry,
   Message,
 } from '../../common/types';
+import { ObservabilityAIAssistantClient } from './client';
 
 export interface IObservabilityAIAssistantClient {
   chat: (options: {
     messages: Message[];
     connectorId: string;
-    functions: Array<Pick<FunctionDefinition['options'], 'name' | 'description' | 'parameters'>>;
-  }) => Promise<IncomingMessage>;
+    functions?: Array<Pick<FunctionDefinition['options'], 'name' | 'description' | 'parameters'>>;
+    stream?: boolean | undefined;
+  }) => ReturnType<ObservabilityAIAssistantClient['chat']>;
```

Comment:
we should probably do the inverse (or maybe just get rid of the interface all together, and just use the class everywhere). But I don't think an interface should be "borrowing" a return type from a concrete implementation

Reply:
Fair enough. Honestly the main reason I did this was because it was proving impossible to type the interface in the same way as the method.

I'll opt for removing the interface.